lib/azure/armrest/armrest_service.rb

```diff
@@ -161,7 +161,11 @@ class << self
         private
 
         def rest_execute(options, http_method = :get)
-          RestClient::Request.execute(options.merge(:method => http_method))
+          options = options.merge(
+            :method => http_method,
+            :url    => Addressable::URI.parse(options[:url]).normalize.to_str
```

Comment:
Well, I'd rather not call extra methods if we don't have to. Could we add some tests?